pulser-core/pulser/math/abstract_array.py

```diff
@@ -271,10 +276,7 @@ def __setitem__(self, indices: Any, values: AbstractArrayLike) -> None:
                 self._process_indices(indices)
             ] = values  # type: ignore[assignment]
         except RuntimeError as e:
-            if (
-                self.is_tensor
-                and cast(torch.Tensor, self._array).requires_grad
-            ):
+            if self.is_differentiable:
```

Comment:
Here for instance I have the feeling that self.requires_grad would be more suited...